### ISSUE TITLE
PackageName:  Kokkos error surfacing in stk

### ISSUE THREAD

**Issue by ajpowelsnl** [trilinos/Trilinos#5056]
## Bug Report
@trilinos/\<PackageName\>

In file included from
/scratch/sierra/sierra_master/stk/stk_ngp/stk_ngp/Ngp.hpp:40:0,
                  from
/scratch/sierra/sierra_master/stk/stk_unit_tests/stk_ngp/howToNgp.cpp:2:
/scratch/sierra/sierra_master/stk/stk_ngp/stk_ngp/NgpField.hpp: In
instantiation of 'bool ngp::StaticField<T>::need_sync_to_device() const
[with T = int]':
/scratch/sierra/sierra_master/stk/stk_ngp/stk_ngp/NgpField.hpp:306:32:
required from 'void ngp::StaticField<T>::sync_to_device() [with T = int]'
/scratch/sierra/sierra_master/stk/stk_ngp/stk_ngp/NgpReductions.hpp:156:5:
   required from 'typename Field::value_type ngp::get_field_min(Mesh&,
Field, const stk::mesh::Selector&) [with Mesh = ngp::StkMeshAdapter;
Field = ngp::StaticField<int>; typename Field::value_type = int]'
/scratch/sierra/sierra_master/stk/stk_unit_tests/stk_ngp/howToNgp.cpp:862:5:
   required from here
/scratch/sierra/sierra_master/stk/stk_ngp/stk_ngp/NgpField.hpp:449:26:
error: 'const FieldDataDualViewType {aka const class
Kokkos::DualView<int***, Kokkos::LayoutRight, void, void>}' has no
member named 'modified_host'; did you mean 'modify_host'?
          return fieldData.modified_host() >
fieldData.modified_device();  // Old Kokkos API
                 ~~~~~~~~~~^~~~~~~~~~~~~
                 modify_host
/scratch/sierra/sierra_master/stk/stk_ngp/stk_ngp/NgpField.hpp:449:54:
error: 'const FieldDataDualViewType {aka const class
Kokkos::DualView<int***, Kokkos::LayoutRight, void, void>}' has no
member named 'modified_device'; did you mean 'modify_device'?
          return fieldData.modified_host() >
fieldData.modified_device();  // Old Kokkos API



### Description

In a Sierra - Trilinos integration, this bug causes Sierra build failures.

### Steps to Reproduce

See Sierra - Trilinos integration procedure communicated in previous bug reports..

**Comment by mhoemmen** [comment]
@ajpowelsnl This is not a Trilinos bug.  Kokkos' interface has changed, and `stk::simd` hasn't kept up.

**Comment by ibaned** [comment]
```cpp
fieldData.modified_device();
```
needs to be replaced with:
```cpp
fieldData.modified_flags(1);
```
This is a bug in STK for relying on deprecated APIs, which will be removed in version 3.0 of Kokkos in less than a year.

**Comment by alanw0** [comment]
More precisely, instead of calling 'modified_device()++' we now call 'modify_device()'.
And instead of 'modified_device() > modified_host()' we now call 'need_sync_host()'. Or vice-versa.

Is something like 'modified_flags(1)' really in the public API?
I don't think we intend to call that.

stk was in the position of needing to build against both old and new versions of trilinos/kokkos. So we had calls to both APIs in #ifdef/#else blocks. Those are now being updated.

**Comment by ibaned** [comment]
@alanw0 you're right, although `modified_flags` is public, it's definitely better to do what you suggest for those two desired use cases.
